### PR TITLE
nds: add BlocksDS port

### PR DESCRIPTION
**Pull request by asiekierka**
The main difference over the previous toolchain is ~120K smaller binaries, faster memcpy/memset, and the many bugfixes accumulated in the last few weeks. However, even though it appears to be stable (perhaps even better) in testing, as BlocksDS's future is less certain - think "Lindy effect" - I think it's worth keeping it as a separate port for the time being.

### CODE REVIEW

**Comment by asiekierka** [comment]
Updated, but temporarily blocked by https://github.com/blocksds/sdk/issues/90

**Comment by asiekierka** [comment]
The final issue with this port that I'm aware of is solved by https://github.com/blocksds/libnds/pull/45

File: Makefile [hunk 1]
```diff
@@ -345,6 +345,14 @@ CFLAGS   += -Wno-strict-prototypes -Wno-pedantic
 CXXFLAGS += -Wno-pedantic
 endif
 
+#
+# As does BlocksDS, currently
+#
+ifeq (${BUILD_NDS_BLOCKSDS},1)
+CFLAGS   += -Wno-strict-prototypes -Wno-pedantic -Wno-undef -Wno-declaration-after-statement
```
[the remark this answers is not in the document]
Comment (asiekierka) [reply Makefile:352]: Re `-Wno-undef`: Apparently, this has been fixed in a later BlocksDS version altogether.

Re `-Wno-declaration-after-statement`:

```
  CC       src/counter.c
In file included from /home/asie/toolchains/blocksds/sdk/libs/libnds/include/nds/arm9/boxtest.h:19,
                 from /home/asie/toolchains/blocksds/sdk/libs/libnds/include/nds.h:134,
                 from src/compat.h:91,
                 from src/board.h:23,
                 from src/counter.c:32:
/home/asie/toolchains/blocksds/sdk/libs/libnds/include/nds/arm9/videoGL.h: In function 'glCallList':
/home/asie/toolchains/blocksds/sdk/libs/libnds/include/nds/arm9/videoGL.h:856:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  856 |     u32 count = *list++;
      |     ^~~
```

It's in BlocksDS. There's an open issue to fix all such warnings, but it's a lot of work. I'll probably fast-track this one, but only for the next release: https://github.com/blocksds/sdk/issues/4

[the remark this answers is not in the document]
Comment (AliceLR) [reply Makefile]: Ah, okay then.

**Pull request merged**